lua/nvim-treesitter/parsers.lua

```diff
@@ -18,6 +18,7 @@ local filetype_to_parsername = {
   pandoc = "markdown",
   rmd = "markdown",
   cs = "c_sharp",
+  ["ocaml.interface"] = "ocaml_interface",
```

Comment:
you mean `["ocamlinterface"] = "ocaml_interface"`

Reply:
`["ocam.linterface"] = "ocaml_interface" was already mentioned below. `ocaml_interface` is the name in the parser C binary

Reply:
The filetype detected by LSP is `ocaml.interface`

```lua
filetypes = { 'ocaml', 'ocaml.menhir', 'ocaml.interface', 'ocaml.ocamllex', 'reason', 'dune' }
```

Reply:
but the filetype that was here before was `ocamlinterface`

Reply:
When I made the chanes I checked that Vim didn't allow for filetypes to have that dot, hence the type in LSP is different from the vim filetype. I'm not sure what's the current situation.

Reply:
Vim will fallback to `ocaml` with `ocaml.interface` has not traditional highlighting. I think this behavior was also implemented for nvim-treesitter though we typically recognize the string as a whole for most parsers.

Should we only keep the new filetype? Or will there be users with `ocamlinterface` configured?

Reply:
Ale is configured to use ocamlinterface, and do the conversion: https://github.com/dense-analysis/ale/blob/a02a4f2811f810877f3c3859cca963f9578ff94a/autoload/ale/handlers/ocamllsp.vim#L7

I opened several PRs to follow that convention, detailed here: https://github.com/ocaml/vim-ocaml/pull/61

IMO changing lspconfig to follow that convention is easier than doing the reverse

Reply:
Closing because the LSP setup does the conversion ocamlinterface => 'ocaml.interface'